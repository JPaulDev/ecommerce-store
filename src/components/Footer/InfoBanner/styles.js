import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  background-color: white;
  display: flex;
  flex-direction: column;
  padding: 20px 0;
  row-gap: 15px;

  @media screen and (min-width: 910px) {
    padding-left: 30px;
    padding-right: 15px;
    flex-direction: row;
  }
`;

export const Text = styled.p`
  flex: 1;
  font-size: 0.75rem;
  margin: 0 15px;
  text-align: center;

  @media screen and (min-width: 910px) {
    text-align: left;
    padding-right: 10px;
    max-width: 350px;
    margin-left: 20px;
    margin-right: auto;
  }
`;

export const InnerContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  row-gap: 8px;

  & :last-child {
    border: 0;
  }
`;

export const LinkContainer = styled.div`
  align-items: center;
  border-right: 1px solid var(--color-grey-10);
  column-gap: 12px;
  display: flex;
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.15rem;
  font-weight: 600;
  padding: 0 15px;
`;

export const Link = styled.a`
  align-items: center;
  color: black;
  column-gap: 5px;
  display: flex;
  font-family: inherit;
  text-decoration: none;

  &:hover {
    color: var(--color-blue-2);
  }
`;
