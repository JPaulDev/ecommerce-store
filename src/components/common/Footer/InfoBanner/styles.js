import styled, { css } from 'styled-components';

const flexMixin = css`
  align-items: center;
  display: flex;
`;

export const Container = styled.div`
  background-color: white;
  flex-direction: column;
  padding: 20px 0;
  row-gap: 15px;
  ${flexMixin}

  @media screen and (min-width: 910px) {
    padding-left: 30px;
    padding-right: 15px;
    flex-direction: row;
  }
`;

export const Text = styled.p`
  flex: 1;
  font-size: var(--font-size-12);
  margin: 0 15px;
  text-align: center;

  @media screen and (min-width: 910px) {
    text-align: left;
    padding-right: 10px;
    max-width: 350px;
    margin-left: 20px;
    margin-right: auto;
  }
`;

export const InnerContainer = styled.div`
  flex-wrap: wrap;
  justify-content: center;
  row-gap: 8px;
  ${flexMixin}

  & :last-child {
    border: 0;
  }
`;

export const LinkContainer = styled.div`
  border-right: 1px solid var(--color-grey-9);
  column-gap: 12px;
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-18);
  font-weight: var(--font-weight-semibold);
  padding: 0 15px;
  ${flexMixin}
`;

export const Link = styled.a`
  color: black;
  column-gap: 8px;
  font-family: inherit;
  text-decoration: none;
  ${flexMixin}

  &:hover {
    color: var(--color-blue-3);
  }
`;
