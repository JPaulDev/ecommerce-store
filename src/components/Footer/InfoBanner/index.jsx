import logo from '../../../assets/images/footer/pc-logo.webp';
import { ReactComponent as Twitter } from '../../../assets/svg/twitter.svg';
import { ReactComponent as Facebook } from '../../../assets/svg/facebook.svg';
import { ReactComponent as Reddit } from '../../../assets/svg/reddit.svg';
import { ReactComponent as Youtube } from '../../../assets/svg/youtube.svg';
import { ReactComponent as Instagram } from '../../../assets/svg/instagram.svg';
import MaterialSymbol from '../../MaterialSymbol';
import * as Styled from './styles';

export default function InfoBanner() {
  return (
    <Styled.Container>
      <img src={logo} width={65} height={72} alt="" />
      <Styled.Text>
        PC Connect International Ltd, Unit 5, Lower Milehouse Ln,
        Newcastle-under-Lyme, Newcastle ST5 9EN
      </Styled.Text>
      <Styled.InnerContainer>
        <Styled.LinkContainer>
          <Styled.Link href="">
            <MaterialSymbol iconSize="26px">directions</MaterialSymbol>
            How To Find Us
          </Styled.Link>
        </Styled.LinkContainer>
        <Styled.LinkContainer>
          <Styled.Link href="">
            <MaterialSymbol iconSize="26px">warehouse</MaterialSymbol>
            Opening Times
          </Styled.Link>
        </Styled.LinkContainer>
        <Styled.LinkContainer>
          Follow Us
          <Styled.Link href="https://www.twitter.com">
            <Twitter
              title="Twitter"
              aria-label="Twitter"
              width={20}
              height={20}
              fill="currentColor"
            />
          </Styled.Link>
          <Styled.Link href="https://www.facebook.com">
            <Facebook
              title="Facebook"
              aria-label="Facebook"
              width={12}
              height={20}
              fill="currentColor"
            />
          </Styled.Link>
          <Styled.Link href="https://www.instagram.com">
            <Instagram
              title="Instagram"
              aria-label="Instagram"
              width={20}
              height={23}
              fill="currentColor"
            />
          </Styled.Link>
          <Styled.Link href="https://www.reddit.com">
            <Reddit
              title="Reddit"
              aria-label="Reddit"
              width={23}
              height={23}
              fill="currentColor"
            />
          </Styled.Link>
          <Styled.Link href="https://www.youtube.com">
            <Youtube
              title="YouTube"
              aria-label="YouTube"
              width={25}
              height={22}
              fill="currentColor"
            />
          </Styled.Link>
        </Styled.LinkContainer>
      </Styled.InnerContainer>
    </Styled.Container>
  );
}
