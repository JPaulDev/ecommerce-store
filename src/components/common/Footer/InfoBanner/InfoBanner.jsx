import Link from 'next/link';
import Image from 'next/future/image';
import logo from '../../../../../public/images/footer/pc-logo.webp';
import Twitter from '../../../icons/Twitter';
import Facebook from '../../../icons/Facebook';
import Reddit from '../../../icons/Reddit';
import Youtube from '../../../icons/Youtube';
import Instagram from '../../../icons/Instagram';
import Directions from '../../../icons/Directions';
import Warehouse from '../../../icons/Warehouse';
import * as Styled from './styles';

export default function InfoBanner() {
  return (
    <Styled.Container>
      <Image src={logo} alt="" />
      <Styled.Text>
        PC Connect International Ltd, Unit 5, Lower Milehouse Ln,
        Newcastle-under-Lyme, Newcastle ST5 9EN
      </Styled.Text>
      <Styled.InnerContainer>
        <Styled.LinkContainer>
          <Link href="/about" passHref>
            <Styled.Link>
              <Directions width={25} height={25} />
              How To Find Us
            </Styled.Link>
          </Link>
        </Styled.LinkContainer>
        <Styled.LinkContainer>
          <Link href="/about" passHref>
            <Styled.Link>
              <Warehouse width={25} height={20} />
              Opening Times
            </Styled.Link>
          </Link>
        </Styled.LinkContainer>
        <Styled.LinkContainer>
          Follow Us
          <Styled.Link
            href="https://www.twitter.com"
            title="Twitter"
            aria-label="Twitter"
          >
            <Twitter width={20} height={20} />
          </Styled.Link>
          <Styled.Link
            href="https://www.facebook.com"
            title="Facebook"
            aria-label="Facebook"
          >
            <Facebook width={12} height={19} />
          </Styled.Link>
          <Styled.Link
            href="https://www.instagram.com"
            title="Instagram"
            aria-label="Instagram"
          >
            <Instagram width={20} height={23} />
          </Styled.Link>
          <Styled.Link
            href="https://www.reddit.com"
            title="Reddit"
            aria-label="Reddit"
          >
            <Reddit width={23} height={23} />
          </Styled.Link>
          <Styled.Link
            href="https://www.youtube.com"
            title="YouTube"
            aria-label="YouTube"
          >
            <Youtube width={25} height={22} />
          </Styled.Link>
        </Styled.LinkContainer>
      </Styled.InnerContainer>
    </Styled.Container>
  );
}
