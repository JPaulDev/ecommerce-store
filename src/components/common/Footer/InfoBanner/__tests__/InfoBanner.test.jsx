import renderer from 'react-test-renderer';
import InfoBanner from '../index';

describe('InfoBanner component', () => {
  it('renders a banner with the company address/information and social media links', () => {
    const tree = renderer.create(<InfoBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
