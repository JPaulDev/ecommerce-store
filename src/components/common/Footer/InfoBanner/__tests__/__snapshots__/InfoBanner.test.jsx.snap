// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBanner component renders a banner with the company address/information and social media links 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px 0;
  row-gap: 15px;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 0.75rem;
  margin: 0 15px;
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  row-gap: 8px;
}

.c2 :last-child {
  border: 0;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-right: 1px solid var(--color-grey-10);
  -webkit-column-gap: 12px;
  column-gap: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Uni Sans',sans-serif;
  font-size: 1.15rem;
  font-weight: 600;
  padding: 0 15px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-column-gap: 8px;
  column-gap: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  color: var(--color-blue-2);
}

@media screen and (min-width:910px) {
  .c0 {
    padding-left: 30px;
    padding-right: 15px;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:910px) {
  .c1 {
    text-align: left;
    padding-right: 10px;
    max-width: 350px;
    margin-left: 20px;
    margin-right: auto;
  }
}

<div
  className="c0"
>
  <img
    alt=""
    data-nimg="future"
    decoding="async"
    height={24}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={24}
  />
  <p
    className="c1"
  >
    PC Connect International Ltd, Unit 5, Lower Milehouse Ln, Newcastle-under-Lyme, Newcastle ST5 9EN
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <a
        className="c4"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="currentColor"
          height={25}
          viewBox="0 0 512 512"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M227.7 11.7c15.6-15.6 40.9-15.6 56.6 0l216 216c15.6 15.6 15.6 40.9 0 56.6l-216 216c-15.6 15.6-40.9 15.6-56.6 0l-216-216c-15.6-15.6-15.6-40.9 0-56.6l216-216zm87.6 137c-4.6-4.6-11.5-5.9-17.4-3.5s-9.9 8.3-9.9 14.8v56H224c-35.3 0-64 28.7-64 64v48c0 13.3 10.7 24 24 24s24-10.7 24-24V280c0-8.8 7.2-16 16-16h64v56c0 6.5 3.9 12.3 9.9 14.8s12.9 1.1 17.4-3.5l80-80c6.2-6.2 6.2-16.4 0-22.6l-80-80z"
          />
        </svg>
        How To Find Us
      </a>
    </div>
    <div
      className="c3"
    >
      <a
        className="c4"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="currentColor"
          height={20}
          viewBox="0 0 640 512"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 488V171.3c0-26.2 15.9-49.7 40.2-59.4L308.1 4.8c7.6-3.1 16.1-3.1 23.8 0L599.8 111.9c24.3 9.7 40.2 33.3 40.2 59.4V488c0 13.3-10.7 24-24 24H568c-13.3 0-24-10.7-24-24V224c0-17.7-14.3-32-32-32H128c-17.7 0-32 14.3-32 32V488c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24zm488 24l-336 0c-13.3 0-24-10.7-24-24V432H512l0 56c0 13.3-10.7 24-24 24zM128 400V336H512v64H128zm0-96V224H512l0 80H128z"
          />
        </svg>
        Opening Times
      </a>
    </div>
    <div
      className="c3"
    >
      Follow Us
      <a
        aria-label="Twitter"
        className="c4"
        href="https://www.twitter.com"
        title="Twitter"
      >
        <svg
          fill="currentColor"
          height={20}
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          />
        </svg>
      </a>
      <a
        aria-label="Facebook"
        className="c4"
        href="https://www.facebook.com"
        title="Facebook"
      >
        <svg
          fill="currentColor"
          height={19}
          viewBox="0 0 320 512"
          width={12}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
          />
        </svg>
      </a>
      <a
        aria-label="Instagram"
        className="c4"
        href="https://www.instagram.com"
        title="Instagram"
      >
        <svg
          fill="currentColor"
          height={23}
          viewBox="0 0 448 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
          />
        </svg>
      </a>
      <a
        aria-label="Reddit"
        className="c4"
        href="https://www.reddit.com"
        title="Reddit"
      >
        <svg
          fill="currentColor"
          height={23}
          viewBox="0 0 512 512"
          width={23}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M201.5 305.5c-13.8 0-24.9-11.1-24.9-24.6 0-13.8 11.1-24.9 24.9-24.9 13.6 0 24.6 11.1 24.6 24.9 0 13.6-11.1 24.6-24.6 24.6zM504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-132.3-41.2c-9.4 0-17.7 3.9-23.8 10-22.4-15.5-52.6-25.5-86.1-26.6l17.4-78.3 55.4 12.5c0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.3 24.9-24.9s-11.1-24.9-24.9-24.9c-9.7 0-18 5.8-22.1 13.8l-61.2-13.6c-3-.8-6.1 1.4-6.9 4.4l-19.1 86.4c-33.2 1.4-63.1 11.3-85.5 26.8-6.1-6.4-14.7-10.2-24.1-10.2-34.9 0-46.3 46.9-14.4 62.8-1.1 5-1.7 10.2-1.7 15.5 0 52.6 59.2 95.2 132 95.2 73.1 0 132.3-42.6 132.3-95.2 0-5.3-.6-10.8-1.9-15.8 31.3-16 19.8-62.5-14.9-62.5zM302.8 331c-18.2 18.2-76.1 17.9-93.6 0-2.2-2.2-6.1-2.2-8.3 0-2.5 2.5-2.5 6.4 0 8.6 22.8 22.8 87.3 22.8 110.2 0 2.5-2.2 2.5-6.1 0-8.6-2.2-2.2-6.1-2.2-8.3 0zm7.7-75c-13.6 0-24.6 11.1-24.6 24.9 0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.1 24.9-24.6 0-13.8-11-24.9-24.9-24.9z"
          />
        </svg>
      </a>
      <a
        aria-label="YouTube"
        className="c4"
        href="https://www.youtube.com"
        title="YouTube"
      >
        <svg
          fill="currentColor"
          height={22}
          viewBox="0 0 576 512"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
